<launch>
  <arg name="node_name" default="equalize_histogram" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show image" />
  <arg name="queue_size" default="3" doc="Specigy queue_size of input image subscribers" />

  <arg name="histogram_equalization_type" default="0" doc="Specify histogram equaliztion methods. 0: Default, 1: Clahe" />

  <arg name="clahe_clip_limit" default="255.0" doc="The maximum allowed field value Clip Limit" />
  <arg name="clahe_tile_size_x" default="300" doc="The maximum allowed field value Threshold X Limit" />
  <arg name="clahe_tile_size_y" default="300" doc="The maximum allowed field value Threshold Y Limit" />

  <arg name="use_opencl" default="true" doc="Enable or disable OpenCL" />

  <!-- equalize_histogram.cpp  -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="equalize_histogram" output="screen">
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="queue_size" value="$(arg queue_size)" />
    <param name="clahe_clip_limit" value="$(arg clahe_clip_limit)" />
    <param name="clahe_tile_size_x" value="$(arg clahe_tile_size_x)" />
    <param name="clahe_tile_size_y" value="$(arg clahe_tile_size_y)" />
    <param name="use_opencl" value="$(arg use_opencl)" />
    <param name="histogram_equalization_type" value="$(arg histogram_equalization_type)" />
  </node>
</launch>
